amical/tests/test_extraction.py

```diff
@@ -12,31 +35,53 @@ def test_add_infos_simulated():
     hdr["DATE-OBS"] = "2021-06-23"
     hdr["TELESCOP"] = "FAKE-TEL"
 
-    # This test is for simulated data only
+    # SimulatedData avoids requiring extra keys in infos
     infos = munch.Munch(orig="SimulatedData", instrument="unknown")
 
     # Add hdr to infos placeholders for everything but hdr
     mf = munch.Munch(pixelSize=1.0)
     infos = _add_infos_header(infos, hdr, mf, 1.0, "afilename", "amaskname", 1)
 
+    # Check that we kept required keys
     assert infos["date-obs"] == hdr["DATE-OBS"]
     assert infos["telescop"] == hdr["TELESCOP"]
-    assert "observer" not in infos  # Keys that are not in hdr should not be in infos
 
+    # Keys that are not in hdr should not be in infos or hdr
+    assert "observer" not in infos
+    assert "observer" not in infos.hdr
 
-def test_add_infos_header_commentary():
+
+@pytest.mark.filterwarnings("ignore: Commentary cards")
+def test_add_infos_header_commentary(commentary_infos):
     # Make sure that _add_infos_header handles _HeaderCommentaryCards from astropy
 
-    # Create a fits header with commentary card
-    hdr = fits.Header()
-    hdr["HISTORY"] = "History is a commentary card"
+    # Convert everything to munch object
+    munch.munchify(commentary_infos)
 
-    # SimulatedData avoids requiring extra keys in infos
-    infos = munch.Munch(orig="SimulatedData", instrument="unknown")
+
+@pytest.mark.xfail(
+    ASTROPY_VERSION < Version("5.0rc"),
+    reason="Munch cannot handle commentary cards for Astropy < 5.0",
+)
+def test_commentary_infos_keep(commentary_infos):
+    assert "HISTORY" in commentary_infos.hdr
```

Comment:
I do not understand why Codecov reports this as not hit. It runs on my machine whatever version of astropy I use. Any clue ? (not needed to get this merged)

Reply:
No sorry... Works on my machine as well, and I don't know much about Codecov